### ISSUE TITLE
Target audience?

### ISSUE THREAD

**Issue by niwaka-ame** [amoghpj/emacs-for-biologists#1]
Thanks Amogh for creating the repo.

If we're going for the style like [awesome-cl](https://github.com/CodyReichert/awesome-cl) (which is great), I'm pretty sure this repo will be useful for biologists who know Emacs well enough.

But if we also want to attract perspective users - by showing that how the tools in the repo might be useful in their research - perhaps we'll need more than a list of repos, but also example gifs/vids? This might be useful for both new comers and power users?

**Comment by niwaka-ame** [comment]
And should we target the repo at general biologists, including both wet-lab/computational, and maybe even people working in relevant disciplines like epidemiologists/MDs? 

Presumably this will allow more repos into the list, and some may be useful for more than just wet-lab biologists.

**Comment by amoghpj** [comment]
> But if we also want to attract perspective users - by showing that how the tools in the repo might be useful in their research - perhaps we'll need more than a list of repos, but also example gifs/vids? This might be useful for both new comers and power users?

This is a good point. I was going for the awesome-cl style to begin with. I think the example gifs is a great idea, but it requires some effort and planning. Do you think it might be possible to integrate these workflow animations into this repository, or would it be better served in a separate repo?

> And should we target the repo at general biologists, including both wet-lab/computational, and maybe even people working in relevant disciplines like epidemiologists/MDs?

I think we should be as broad as possible with the definition of "biologist". It might be most useful for people who are already familiar with the command line, i.e. bioinformatics types, but I hope to expand to interfacing with things like ImageJ and Pymol. I also have a small package for defining differential equations models which can hook into various python backends that might be useful for modelers in general, and epidemiologists/systems biologists in particular.

But I do think that there is value in also accumulating some utilities focused towards wet-lab biologists. I think it is best to start of as general as possible, and then add more subsections as the fields become more specialized. 

My current workflow is very org-mode centered. Making a workflow video will assume that the audience is familiar with org-mode to some extent. I understand that a lot of biologists are familiar with Rstudio or Jupyter. Do you have any recommendations of workflow videos for biologists from these communities? 

Finally, would you be interested in making a wishlist section for things that you wish had better emacs integration? Some things that come to mind are viewing sequence annotations (example: rendering genbank files), rendering pdb structures, and querying databases (much like your SGD interface).

**Comment by niwaka-ame** [comment]
> I think the example gifs is a great idea, but it requires some effort and planning. Do you think it might be possible to integrate these workflow animations into this repository, or would it be better served in a separate repo?

I think it might be better to include both in the same repo, but perhaps in a different org file (to avoid a tsunami of info).  This means we can add an entry and its corresponding example in a single commit. 

> I think we should be as broad as possible with the definition of "biologist". It might be most useful for people who are already familiar with the command line, i.e. bioinformatics types, but I hope to expand to interfacing with things like ImageJ and Pymol.

Agree. I spotted how people use Pymol within org-babel during the Emcas conf: https://emacsconf.org/2021/talks/molecular/  - and I realised that the form of the package isn't necessarily Emacs Lisp. 

> But I do think that there is value in also accumulating some utilities focused towards wet-lab biologists. I think it is best to start of as general as possible, and then add more subsections as the fields become more specialized.

Agree. Perhaps 3 subsections would suffice: computational, wet-lab and general info querying (for all biologists)?

> My current workflow is very org-mode centered. Making a workflow video will assume that the audience is familiar with org-mode to some extent. I understand that a lot of biologists are familiar with Rstudio or Jupyter. Do you have any recommendations of workflow videos for biologists from these communities?

Mine is extremely org-mode centred too, and I didn't watch a lot of workflow videos specifically for biologists. But I presume those would belong to a blog post but not within this repo (but we can include links)?  

We can perhaps have another subsection for non-package stuff - that is, probably workflows. In this case the audience will be even broader - ways of note taking, agenda management and literature management should be applicable to most academics. 

> Finally, would you be interested in making a wishlist section for things that you wish had better emacs integration? Some things that come to mind are viewing sequence annotations (example: rendering genbank files), rendering pdb structures, and querying databases (much like your SGD interface).

I would love to have a general info query system, given that Emacs's fuzzy search is so good, and that it's so easy to create interactive UIs using Emacs Lisp. This should save me a lot of time without leaving Emacs, searching and mouse-clicking. 

I've been prototyping a bigger, gene-centric query framework that in theory works for more organisms (and `sgd-lookup.el` will be part of the S. cerevisiae-specific functions), and hopefully I'll publish it on GitHub by this weekend. Let me know if you're interested? 

By the way, here's a paper (and a major mode) I came acorss about handling FASTA and GenBank files: https://academic.oup.com/bioinformaticsadvances/article/1/1/vbab019/6374492  and we should probably add it to this repo.

**Comment by amoghpj** [comment]
> I think it might be better to include both in the same repo, but perhaps in a different org file (to avoid a tsunami of info)

I totally agree.

>  I spotted how people use Pymol within org-babel 

Good find! I've added this to the document.

> Perhaps 3 subsections would suffice: computational, wet-lab and general info querying (for all biologists)?

I've organized the file in this layout. Feel free to add more links and code snippets! I think once we start adding more resources we can figure out if we need more, or less structure.

>I've been prototyping a bigger, gene-centric query framework that in theory works for more organisms (and sgd-lookup.el will be part of the S. cerevisiae-specific functions), and hopefully I'll publish it on GitHub by this weekend. Let me know if you're interested?

I am very interested in this! I have been developing ad hoc functions that wrap entrezpy to query NCBI. Similarly, I've been meaning to right a helm interface to work with SRA files to interface with RNA seq datasets. I think there is so much that can be done. I am very interested in your progress on this, let me know if I can contribute in any way?

> By the way, here's a paper (and a major mode) I came acorss about handling FASTA and GenBank files: https://academic.oup.com/bioinformaticsadvances/article/1/1/vbab019/6374492 and we should probably add it to this repo.

Done! I am quite surprised to see an emacs mode published in a journal! Maybe we can put together one of those PLOS "10 tips to make the best use of emacs for biology" guides at some point haha.

**Comment by niwaka-ame** [comment]
> I've organized the file in this layout.

Many thanks, amogh! Apologies - I didn't really you've had another org file in addition to the readme. Perhaps a link from the readme to the list would help people find? We could even highlight some of those and give 1 or 2 examples in the readme once we have a more complated list, just to attract people.

> I am very interested in your progress on this, let me know if I can contribute in any way?

It'd be great if you're happy to join! Lots of stuff can potentially be done, especially Entrez and KEGG APIs. 
My current idea is a package with a curated collection of local data (e.g. yeast deletion collection info) + corresponding querying functions + querying external databases through Elisp API wrappers, for each species. 

Everything will be interactive and new users don't need to know org-mode or coding unless they want to hack :) 

I'll publish it tonight when I get home, so you can see the rough idea ;) we can brainstorm in that repo too!

> PLOS "10 tips to make the best use of emacs for biology" guides at some point haha.

Yeah haha - that's the way to approach biologists. The majority of biologists aren't GitHub users I guess?

**Comment by amoghpj** [comment]
Good point. I've added a link to the readme. We can even consider moving everything to the readme once we have a better structure. 

Very excited about your project, looking forward to it!

**Comment by niwaka-ame** [comment]
Hey! I've published my repo here: https://github.com/niwaka-ame/emacs-gene-tools
